internal/pkg/propagation/propagation.go

```diff
@@ -258,14 +258,13 @@ func (prop *Propagation) visitCall(call *ssa.Call, maxInstrReached map[*ssa.Basi
 	// Source methods that return tainted values regardless of their arguments should be identified by the fieldpropagator analyzer.
 	if recv := call.Call.Signature().Recv(); recv != nil && sourcetype.IsSourceType(prop.config, prop.taggedFields, recv.Type()) {
 		visitingFromArg := false
-		// Interface types cannot be sources. If the receiver is not a source, the
-		// above condition will be false, so this code won't be executed.
-		// When the receiver's type is statically known (it isn't an interface type),
-		// it will be the first element of the Args slice.
+		// If the receiver is not statically known (it has interface type) and the
+		// method has no arguments, Args will be empty.
 		if len(call.Call.Args) == 0 {
-			fmt.Printf("call.Call.Args was empty; please report this issue")
 			return
 		}
+		// If the receiver's type is statically known,
+		// it will be the first element of the Args slice.
```

Comment:
What does this imply for, e.g.,

```golang
func Foo(si SourceInterface, s Source) {
  t := si.TaintingMethod(s)
  Sink(t)
}
```

From the comments here, I'd think that the dynamically dispatched `TaintingMethod` wouldn't get `si` as an arg?  Which means taint propagate correctly in the above case?

Reply:
I assume you meant, *in*correctly, as that is indeed the case. Good catch, we should only skip the first argument when the receiver is statically known. I've added tests verifying this and fixed the bug.

Reply:
Also, I note that the `fieldpropagator`'s functionality should perhaps be extended to include non-struct types, as currently it will only identify method propagators that return a field.

Reply:
I did drop my negative.  Way to read what I mean, not what I say.